### ISSUE TITLE
Allowing competitors to edit registrations

### ISSUE THREAD

**Issue by KitClement** [thewca/worldcubeassociation.org#941]
Currently, once a registration is approved, the competitor is unable to edit it.  This is very safe, as if the competitor paid for a certain event list and then were able to change this after it were approved, organizers would have a difficult time keeping track of who paid and who didn't.  It would be nice to have some kind of flow for competitors to be able to edit registrations, and then be sent back to a waiting area, but one different from the waiting list.  The image below shows how I perceive this from the organizer's perspective:

![image](https://cloud.githubusercontent.com/assets/6166412/19610057/470ee20e-978f-11e6-8442-ad59935afc36.png)

We would have two sections of the waiting list, changed registrations and new registrations.  New registrations would work as it does now, but changed registrations would contain previously approved registrations that had a change on the user's end that requires approval.  Registrations from this section would still display publicly, but with the old events shown.  The organizer would be able to see the old/new registrations together as shown in the picture with some sort of color coding to indicate additions and removals.  The organizer could then approve the registration once additional payments are cleared.

For competitions that don't have per-payment event systems, they could opt-in to an option where registration edits by users go live immediately.

Forgive me if this is too much to implement, or if this schema doesn't align with any plans regarding the implementation of payments into the WCA, but I just thought I would propose the idea I had.

EDIT: blurred out birthdays

**Comment by jfly** [comment]
This is a lovely mock up, thanks Kit! I like the flow you've described.

@pingskills is our only guy currently working on registration stuff, and he's focusing on payments right now.

**Comment by Mollerz** [comment]
As an aside to this, allow people to delete their own registrations once confirmed. It's quite a pain to have to go through and find the people who want their registration removed, when it would make sense they could do it themselves.

**Comment by BiscuitCubed** [comment]
We'd have to decide what to do with payments once they're added for deleting registrations. Some people would have a once paid, it's paid stance, while some people may want refund part or all of the fee.

**Comment by bugybunny** [comment]
What’s the status on this? I wanted this feature so many times. I’ll explain from my view as competitor. 

When a registration opens I just register for all my events I’m normally doing. Sometimes, I decide that I don’t want to go on some days (for multi day comps) because it’s far away and I decide it’s not worth it only to go for Pyra and 4x4 as an example. So I want to take them off the list but I can’t . And registering only for events I really really want to do is also not satisfying as the schedule might change and I want to compete in more events.

So I end up writing the organizer because I don’t want them to have extra work in printing the score sheets, placing me in a group or even write me up for staffing when I don’t show up on this day. Either way generates work for the organizer that does not have to be there if this feature would be implemented.

I can think of a few problems (I’m sure other people will name a few more and tell me that this is not feasible :p) and have given them some thoughts:

- Only allow it until registration is open obviously
- Only allow this for comps that do not have a per-event-fee but only a fee for the whole competition. I don’t know what information the WCA has about a comp exactly. So dunno if there are still comps that would collect the fee during the registration at the comp itself instead of over the website. This would maybe cause some problems.
- I had one more but forgot about it while writing the other :(

I have no idea about stripe but is the payment immediately? If not, charge only when the registration closes and this would also work for comps with per-event-fees. Or do you think this would encourage people even more to just register for a comp when it pops up and then decide later if they don’t want to attend and take away spots for other people?

**Comment by EdHollingdale** [comment]
Is there a further update on this? In Australia we have competition contact centrally and this would save myself and the other person managing the list several hours a week.

A simpler approach to Kit's suggestion may be just having a box on the new competition form that could be selected "Allow competitors to edit events while registration is open" (so competitions with per-event fees or event combination restrictions would chose not to enable it).

I can't emphasise how much time this would save organisers!

**Comment by jfly** [comment]
@EdHollingdale, we have a lot of work to do to improve our registration system. I don't think things are at a point right now where we can just start chipping away at work. I think someone needs to meditate on the state of things for a while, and think about what exactly we want to build, and how we're going to get there. That is a big project that nobody on the WST currently has time to work on. I'm hoping that @ronaldmansveld will be able to sit down and think about how to revamp our registration platform once he frees up from wca-live work.

**Comment by EdHollingdale** [comment]
Without meaning to be a pain I'll just keep bumping this every now and then so it is clear where in a list of priorities this should sit, hopefully quite high!

I got bored during an FMC attempt I wasn't competing in so I did some calculations. Speedcubing Australia has a centralised mailing list for all competitions. During the month of March we had 73 event change requests through the mailing list plus another 25 sent directly to me. Conservatively we'll say 100 event changes per month in Australia, which holds ~3% of all comps. Saying 2 minutes per event change (finding the comp, finding their name, changing their registration, writing an email back) that is 100x2/(0.03x60) = **111 hours collectively spent per month** by organisers and delegates just changing people's events. I'm not saying that allowing competitors to edit registrations would eliminate all of those emails, but it would put a big dent in that time.

**Comment by KerrieJarman** [comment]
Is there anyone that can start working on this? PLEASE??  I am at the stage now where I am willing to pay real money out of my own pocket to fix this, if it means I don't have to keep doing event changes for users. 
 
@EdHollingdale did some great calculations a while back on how much time is WASTED on changing events on behalf of users - a task they really should be able to do themselves and it has only got worse. Here in Australia all our competitions are centralised and we get to see all the event change requests, at the moment there are hundreds every month!

**Comment by saranshgrover** [comment]
I believe @timreyn is working on this?

**Comment by timreyn** [comment]
I created #6305 for this.  It adds an option to the competition form; the default is not allowing edits after registration is opened (i.e. the current behavior).
